File: CloudMusic/src/main/java/com/ken/android/CloudMusic/Dialog/CollectDialog.java
package com.ken.android.CloudMusic.Dialog;

import android.app.Dialog;
import android.content.Context;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.annotation.StyleRes;
import android.support.v7.app.AlertDialog;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.ken.android.CloudMusic.DBHelper.ListsInfoDao;
import com.ken.android.CloudMusic.FilesRead.ListsInfo;
import com.ken.android.CloudMusic.FilesRead.MusicInfo;
import com.ken.android.CloudMusic.R;

import org.xutils.view.annotation.ContentView;
import org.xutils.view.annotation.ViewInject;
import org.xutils.x;

import java.util.List;
import java.util.zip.Inflater;

/**
 * Created by axnshy on 16/8/11.
 */
public class CollectDialog extends Dialog{

    private List<ListsInfo> mList;
    private Context context;
    private RecyclerView recycler;
    private MusicInfo musicInfo;
    private Adapter mAdapter;

    public CollectDialog(Context context, List<ListsInfo> mList,MusicInfo music) {
        super(context);
        this.mList = mList;
        this.context = context;
        this.musicInfo=music;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_collect);
        recycler= (RecyclerView) findViewById(R.id.recycler_dialog_list);
        mAdapter=new Adapter(context,mList);
        mAdapter.setOnClickListener(new Adapter.OnClickListener() {
            @Override
            public void onItemClick(View view, int position) {
                ListsInfoDao.getDAO().addMusicToList(context,musicInfo,mList.get(position).getListId());
                dismiss();
            }
            @Override
            public void onItemLongClick(View view, int position) {
            }
        });
        recycler.setAdapter(mAdapter);
        LinearLayoutManager manager=new LinearLayoutManager(context,LinearLayoutManager.VERTICAL,false);
        recycler.setLayoutManager(manager);
    }


}
class Adapter extends RecyclerView.Adapter<MViewHolder>{

    private LayoutInflater mInflater;
    private Context context;
    private List<ListsInfo> mList;
    public Adapter(Context context,List<ListsInfo> List){
        this.context=context;
        this.mList=List;
        mInflater = LayoutInflater.from(context);
    }

    OnClickListener mListener;

    public interface OnClickListener {
        void onItemClick(View view, int position);

        void onItemLongClick(View view, int position);
    }

    public void setOnClickListener(OnClickListener listener){
        this.mListener=listener;
    }


    @Override
    public MViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = mInflater.inflate(R.layout.dialog_list_item, parent, false);
        MViewHolder viewHolder = new MViewHolder(view);
        return viewHolder;
    }
    @Override
    public void onBindViewHolder(final MViewHolder holder, final int position) {
        Log.w("TAG","image path ----------   "+mList.get(position).getBackgroundPath());
        holder.listAvatarImg.setImageBitmap(BitmapFactory.decodeFile(mList.get(position).getBackgroundPath()));
        holder.listNameTx.setText(mList.get(position).getListName());
        if(mListener!=null) {
            holder.itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    mListener.onItemClick(holder.itemView,position);
                }
            });
            holder.itemView.setOnLongClickListener(new View.OnLongClickListener() {
                @Override
                public boolean onLongClick(View view) {
                    mListener.onItemLongClick(holder.itemView,position);
                    return false;
                }
            });
        }
    }

    @Override
    public int getItemCount() {
        return mList.size();
    }
}
class MViewHolder extends RecyclerView.ViewHolder{

//    @ViewInject(R.id.iv_dialog_list_avatar)
    ImageView listAvatarImg;
//    @ViewInject(R.id.tv_dialog_list_name)
    TextView listNameTx;

    public MViewHolder(View itemView) {
        super(itemView);
        listAvatarImg= (ImageView) itemView.findViewById(R.id.iv_dialog_list_avatar);
        listNameTx= (TextView) itemView.findViewById(R.id.tv_dialog_list_name);
    }
}

File: SmartPlay/src/main/java/com/luffy/smartplay/ui/dialog/CollectDialog.kt
package com.luffy.smartplay.ui.dialog

import android.app.Dialog
import android.content.Context
import android.graphics.BitmapFactory
import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.app.DialogCompat
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.luffy.smartplay.db.bean.AlbumData
import com.luffy.smartplay.db.bean.MusicData
import com.luffy.smartplay.databinding.DialogCollectBinding
import com.luffy.smartplay.databinding.DialogListItemBinding
import com.luffy.smartplay.db.MyDatabase
import com.luffy.smartplay.db.repo.MusicRepository
import com.luffy.smartplay.ui.base.BaseBindingAdapter
import com.luffy.smartplay.ui.base.BindingViewHolder
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch

/**
 * Created by axnshy on 16/8/11.
 */
class CollectDialog(var data: List<AlbumData>, val music: MusicData,
                    context: Context,
                    themeResId: Int
) : Dialog(context, themeResId) {
    private var recycler: RecyclerView? = null
    private var mAdapter: Adapter? = null
    private lateinit var viewBinding : DialogCollectBinding
    override fun onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        setContentView(DialogCollectBinding.inflate(layoutInflater).apply {
            viewBinding = this
        }.root)

        mAdapter = Adapter(data)
        mAdapter?.eventListener = BaseBindingAdapter.EventListener().apply {
            itemClickListener = {
                    view,position->
                GlobalScope.launch {
                    MusicRepository.insertToAlbum(data[position].albumId,music)
                }
                dismiss()
            }
        }
        viewBinding.recyclerDialogList.adapter = mAdapter
        viewBinding.recyclerDialogList.layoutManager  = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)

    }
}

internal class Adapter(val list: List<AlbumData>) :
    BaseBindingAdapter<DialogListItemBinding>() {

    override fun onBindViewHolder(holder: BindingViewHolder<DialogListItemBinding>, position: Int) {
        super.onBindViewHolder(holder, position)
        holder.holder.apply {
            ivDialogListAvatar.setImageBitmap(BitmapFactory.decodeFile(list[position].backgroundPath))
            tvDialogListName.text = list[position].albumName
        }
    }


    override fun createViewHolder(
        layoutInflater: LayoutInflater,
        parent: ViewGroup,
        viewType: Int
    ): DialogListItemBinding {
        return DialogListItemBinding.inflate(layoutInflater,parent,false)
    }

    override fun getItemCount(): Int {
        return list.size
    }
}